right-click-context.el

```diff
@@ -207,15 +207,26 @@ You probably want to just add follows code to your .emacs file (init.el).
     (call-interactively #'mouse-set-point))
   (right-click-context-menu))
 
+;;;###autoload
+(defun right-click-context-click-menu-popup ()
+  "Open a new right click context menu at the new mouse position."
+  (interactive)
+  (when (memq this-command '(right-click-context-click-menu))
+    (popup-delete (nth (1- (length popup-instances)) popup-instances))
+    (call-interactively #'right-click-context-click-menu)))
+
 ;;;###autoload
 (defun right-click-context-menu ()
   "Open Right Click Context menu."
   (interactive)
-  (let ((value (popup-cascade-menu (right-click-context--build-menu-for-popup-el (right-click-context--menu-tree) nil))))
-    (when value
-      (if (symbolp value)
-          (call-interactively value t)
-        (eval value)))))
+  (let ((ori-popup-menu-keymap (copy-sequence popup-menu-keymap)))
```

Comment:
The `ori-` prefix is not required. There is no need to back up the original values. Because Emacs Lisp is a dynamic scope, the value bound by the variable in let is valid only in that let expression. You can evaluate `popup-menu-keymap` to ensure that the original keymap is uncontaminated.

```el
diff --git a/right-click-context.el b/right-click-context.el
index d30c8c8..05401af 100644
--- a/right-click-context.el
+++ b/right-click-context.el
@@ -219,14 +219,13 @@ You probably want to just add follows code to your .emacs file (init.el).
 (defun right-click-context-menu ()
   "Open Right Click Context menu."
   (interactive)
-  (let ((ori-popup-menu-keymap (copy-sequence popup-menu-keymap)))
+  (let ((popup-menu-keymap (copy-sequence popup-menu-keymap)))
     (define-key popup-menu-keymap [mouse-3] #'right-click-context-click-menu-popup)
     (let ((value (popup-cascade-menu (right-click-context--build-menu-for-popup-el (right-click-context--menu-tree) nil))))
       (when value
         (if (symbolp value)
             (call-interactively value t)
-          (eval value))))
-    (setq popup-menu-keymap ori-popup-menu-keymap)))
+          (eval value))))))
```

Reply:
Fixed! :D